c2cgeoform/routes.py

```diff
@@ -13,19 +11,25 @@ def table_pregenerator(request, elements, kw):
     def rec_with_pregenerator(route, pattern):
         config.add_route(route, pattern, pregenerator=table_pregenerator)
 
-    global tables
+    tables = []
     for key, mapper in models:
         ca_config = getattr(mapper, '__colanderalchemy_config__', {})
         table = {}
+        # key stand for url path, sometimes designated as table (in match dict)
         table['key'] = key
-        table['name'] = mapper.__tablename__
-        table['title'] = ca_config.get('title', table['name'])
-        table['plural'] = ca_config.get('plural', table['title'])
+        table['title'] = ca_config.get('title', mapper.__tablename__)
+        table['plural'] = ca_config.get('plural', mapper.__tablename__)
         tables.append(table)
 
-    table_regex = '[{}]'.format('|'.join([table['key'] for table in tables]))
+    table_regex = '({})'.format(')|('.join([table['key'] for table in tables]))
```

Comment:
I would prefer '|'.join('({})'.format([table['key']) for table in tables])
for understanding

Reply:
+1